src/modules/users/users.service.ts

```diff
@@ -1,28 +1,129 @@
+    });
+    if (!account || !account.profile) {
+      throw new NotFoundException('Account not found');
+    }
+
+    return account.profile;
+  }
+
+  async getUserWishlist(accountId: string) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+    });
+    if (!account) {
+      throw new NotFoundException('Account not found');
+    }
+
+    const userWishlist = await this.wishlistRepository.find({
+      where: {
+        accountId,
+      },
+    });
+
+    return userWishlist;
+  }
+
+  async updateProfile(accountId: string, updateProfileDto: UpdateProfileDto) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+      relations: ['profile'],
+    });
+    if (!account || !account.profile) {
+      throw new NotFoundException('Account not found');
+    }
+
+    Object.assign(account.profile, updateProfileDto);
+
+    const savedProfile = await this.userProfileRepository.save(account.profile);
+    return savedProfile;
   }
 
-  findOne() {
-    return `This action returns user`;
+  async updateProfileAvatar(accountId: string, avatarUrl: string) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+      relations: ['profile'],
+    });
+    if (!account || !account.profile) {
+      throw new NotFoundException('Account not found');
+    }
+
+    account.profile.image = avatarUrl;
+
+    const savedProfile = await this.userProfileRepository.save(account.profile);
+    return savedProfile;
   }
 
-  update() {
-    return `This action updates user`;
+  async toggleUserRole(accountId: string) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+    });
+    if (!account) {
+      throw new NotFoundException('Account not found');
+    }
+
+    account.role = account.role === Roles.Admin ? Roles.User : Roles.Admin;
+
+    const updatedUser = await this.userAccountRepository.save(account);
+    return updatedUser;
   }
 
-  remove() {
-    return `This action removes user`;
+  async deleteAccount(deleterId: string, deletingAccountId: string) {
+    const deleter = await this.userAccountRepository.findOne({
+      where: { id: deleterId, isDeleted: false },
+    });
+    if (deleter.role != Roles.Admin && deleterId != deletingAccountId) {
+      throw new ForbiddenException('Only owner or admin can delete account');
+    }
+
+    const deletingUser = await this.userAccountRepository.findOne({
+      where: { id: deletingAccountId, isDeleted: false },
+    });
+    if (!deletingUser) {
+      throw new NotFoundException('User not found or already deleted');
+    }
+
+    deletingUser.isDeleted = true;
+    const deletedUser = await this.userAccountRepository.save(deletingUser);
+
+    const { ...result } = deletedUser;
+    delete result.password;
+
+    return result;
```

Comment:
Actually, there is no need to return the deleted account.

Reply:
Fixed